spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc

```diff
@@ -553,36 +554,145 @@ public class WordCountProcessorApplication {
 	public KStream<?, String> process(KStream<?, String> input) {
 		return input
-				.flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
-				.map((key, word) -> new KeyValue<>(word, word))
-				.groupByKey(Serdes.String(), Serdes.String())
-				.count(TimeWindows.of(5000), "store-name")
-				.toStream()
-				.map((w, c) -> new KeyValue<>(null, "Count for " + w.key() + ": " + c));
-	}
+                .flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
+                .groupBy((key, value) -> value)
+                .windowedBy(TimeWindows.of(5000))
+                .count(Materialized.as("WordCounts-multi"))
+                .toStream()
+                .map((key, value) -> new KeyValue<>(null, new WordCount(key.key(), value, new Date(key.window().start()), new Date(key.window().end()))));
+    }
 
 	public static void main(String[] args) {
 		SpringApplication.run(WordCountProcessorApplication.class, args);
 	}
 ----
 
-If you build it as Spring Boot runnable fat jar, you can run the above example in the following way:
+If you build it as a Spring Boot uber jar, you can run the above example in the following way:
 
 [source]
 ----
 java -jar uber.jar  --spring.cloud.stream.bindings.input.destination=words --spring.cloud.stream.bindings.output.destination=counts
 ----
 
-This means that the application will listen from the incoming Kafka topic words and write to the output topic counts.
+This means that the application will listen from the incoming Kafka topic `words` and write to the output topic `counts`.
 
 Spring Cloud Stream will ensure that the messages from both the incoming and outgoing topics are bound as KStream objects.
-As one may observe, the developer can exclusively focus on the business aspects of the code, i.e. writing the logic required in the processor rather than setting up the streams specific configuration required by the Kafka Streams infrastructure.
-All those boilerplate is handled by Spring Cloud Stream behind the scenes.
+Applications can exclusively focus on the business aspects of the code, i.e. writing the logic required in the processor rather than setting up the streams specific configuration required by the Kafka Streams infrastructure.
+All such interactions are handled by the framework.
+
+=== Message conversion in Spring Cloud Stream Kafka Streams applications
+
+If the following property is set (default is false), the framework skips all message conversions on the outbound (producer) side and it is then done by Kafka itself.
+
+`spring.cloud.stream.bindings.output.producer.useNativeEncoding`.
+
+Similarly, if the following property is set (default is false), any message conversion is skipped on the inbound (consumer) side and natively done by Kafka.
+
+`spring.cloud.stream.bindings.input.consumer.useNativeDecoding`.
+
+When native encoding is disabled, then the messages on the outbound are converted by Spring Cloud Stream using the provided contentType.
```

Comment:
I would reword it as "...provided `contentType` header"?

Reply:
And basically every time you mention header name, property name, or value such as true/false, i would put them in some type of quotes. Although it's up to you

Reply:
will do.